src/figuresClasses.ts

```diff
@@ -1,19 +1,105 @@
-export interface Figure {
+type Shape = 'triangle' | 'circle' | 'rectangle';
+type Color = 'red' | 'green' | 'blue';
 
+export interface Figure {
+  shape: string;
+  color: Color;
+  radius?: number;
+  sides?: number[];
+  getArea: () => number;
 }
 
-export class Triangle {
+export class Triangle implements Figure {
+  public shape: Shape = 'triangle';
 
-}
+  public sides: number[] = [];
+
+  constructor(
+    public color: Color,
+    ...sides: number[]
+  ) {
```

Comment:
Not the best place to use the rest operator

```suggestion
  constructor(
    public color: Color,
    public a: number,
    public b: number,
    public c: number
  ) {
```